src/pages/install/source.md

```diff
@@ -0,0 +1,145 @@
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Building RGBDS from source
+
+RGBDS is open-source, which means you can always compile it yourself, for example if no pre-built solution is available to you, or if you want a specific version.
+
+RGBDS aims to be relatively painless to compile, keeping its dependencies to a minimum:
+- A C11-compatible compiler
+- `libpng` (optional, only required for RGBGFX)
+
+:::caution
+
+Some of the build steps vary between systems, so the instructions provided here are somewhat generic.
+Windows works largely differently from everything else, so some expertise is expected if attempting this on Windows.
+
+:::
+
+## 1. Get the source code
+
+You first need to get the source files to be compiled, using one of the methods below.
+
+<Tabs>
+<TabItem value="snapshot" label="Downloading source snapshots">
+
+1. [Pick a release](/docs).
+2. Under the "GitHub links" heading, follow the `release page` link.
+3. Go to "**Assets**" at the bottom of that page.
+4. Download any of the last three listed files (`rgbds-<version>.tar.gz`, "Source code (`zip`)", or "Source code (`tar.gz`)").
+5. Extract ("unzip") the file.
+
+</TabItem>
+<TabItem value="git" label="Using Git">
+
+1. [Clone the repository](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository).
+2. Then, run `git checkout <version>` (e.g. `git checkout v0.4.1`).
+3. You're set!
+
+</TabItem>
+</Tabs>
+
+## 2. Build
+
+The following are required to build RGBDS; most of those should already be installed on your system.
+
+ - `make` and/or `cmake`, depending on what method you choose below
+ - The `bash` shell
+ - A C compiler, typically `gcc` or `clang`
+ - `bison` (or `yacc`/`byacc` prior to RGBDS 0.5.0)
+ - `pkg-config` (unless using `cmake`)
+
+If you are building RGBGFX, the following are also required:
+
+ - A C++ compiler, typically `g++` or `clang`
+ - `libpng`'s development headers. For Linux users: the package is typically called [`libpng-dev`](https://packages.ubuntu.com/focal/libpng-dev), [`libpng-devel`](https://software.opensuse.org/package/libpng16-devel-64bit), or sometimes plainly [`libpng`](https://www.archlinux.org/packages/extra/x86_64/libpng/). Additionally, a `16` or `1.6` may be present after `libpng` ([example](https://tracker.debian.org/pkg/libpng1.6)).
+
+Two build systems are supported: plain old `make`, and CMake.
+Both can be used on most systems, but Windows users will almost certainly be unable to use `make`.
```

Comment:
This is false, 'make' can be used on Windows. I've been using it from here https://github.com/skeeto/w64devkit

I'm trying rgbds on Windows right now so I don't have to waste memory on wsl, and it seems to work other than the one liner command from here https://rgbds.gbdev.io/docs/v0.6.1/rgbds.7 it seems to fail due to the pipes.

What was your specific problem with Windows @ISSOtm? I know there are many problems though lol

Reply:
```suggestion
Two build systems are supported: plain old `make`, and CMake.
```

Reply:
Yes, I know `make` can be used on Windows, hence the "almost". The second line was added after several Windows users came confused as to which they should pick, and/or after struggling to get Make *and* the corollary-required Coreutils.

Reply:
> I'm trying rgbds on Windows right now so I don't have to waste memory on wsl, and it seems to work other than the one liner command from here https://rgbds.gbdev.io/docs/v0.6.1/rgbds.7 it seems to fail due to the pipes.

dont you "just" need to use bash for that?

Reply:
> dont you "just" need to use bash for that?

I said pipes but I was actually referring to whatever the '-' dashes are, it doesn't seem to work on both git bash and cmd/powershell. `error: -: Cannot read node #0's iter #43: Unexpected end of file`. Changing line endings doesn't seem to make it work either.

Reply:
`-` instructs to use stdin/out, which should indeed be pipes. Though Windows tends to mangle binary data going through those, yay!